src/System.Diagnostics.Process/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -67,16 +69,10 @@ internal static PerformanceCounterLib GetPerformanceCounterLib(string machineNam
             else
                 machineName = machineName.ToLowerInvariant();
 
-            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new Dictionary<string, PerformanceCounterLib>());
+            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new ConcurrentDictionary<(string, string), PerformanceCounterLib>());
+            (string machineName, string lcidString) libraryIdentity = (machineName, lcidString);
 
-            string libraryKey = machineName + ":" + lcidString;
-            PerformanceCounterLib library;
-            if (!PerformanceCounterLib.s_libraryTable.TryGetValue(libraryKey, out library))
-            {
-                library = new PerformanceCounterLib(machineName, lcidString);
-                PerformanceCounterLib.s_libraryTable[libraryKey] = library;
-            }
-            return library;
+            return PerformanceCounterLib.s_libraryTable.GetOrAdd(libraryIdentity, (k, arg) => new PerformanceCounterLib(arg.machineName, arg.lcidString), libraryIdentity);
```

Comment:
Now that the key is itself the value tuple, since the key is passed into the delegate you can revert back to using the original overload and don't need to pass the same tuple in twice:
```C#
var table = LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new ConcurrentDictionary<(string machineName, string lcidString), PerformanceCounterLib>());
return table.GetOrAdd((machineName, lcidString), k => new PerformanceCounterLib(k.machineName, k.lcidString));
```